src/plugins/intel_gpu/tests/test_cases/gather_gpu_test.cpp

```diff
@@ -8,10 +8,334 @@
 #include <intel_gpu/primitives/gather.hpp>
 
 #include <cstddef>
+#include <array>
 
 using namespace cldnn;
 using namespace ::tests;
 
+std::ostream& operator<<(std::ostream& os, FLOAT16 x) {
+    return os << float(x);
+}
+
+template <class T>
+int dim(const T& a) {
+    int ret = a.size();
+    while (ret - 1 >= 0 && a[ret - 1] == 1)
+        ret--;
+    return ret;
+};
+
+template <class T>
+int mult(const T& a) {
+    return std::accumulate(a.begin(), a.end(), 1, [](int x, int y) {
+        return x * y;
+    });
+};
+
+using gather8_test_4d_param = std::tuple<int,                  // batch_dim
+                                         int,                  // axis
+                                         format::type,         // format of input0
+                                         format::type,         // format of input1
+                                         std::array<int, 4>,   // bfyx0
+                                         std::array<int, 4>>;  // bfyx1
+using gather8_test_5d_param = std::tuple<int,                  // batch_dim
+                                         int,                  // axis
+                                         format::type,         // format of input0
+                                         format::type,         // format of input1
+                                         std::array<int, 5>,   // bfzyx0
+                                         std::array<int, 5>>;  // bfzyx1
+
+// index 0 is input0
+// index 1 is input1
+// index 2 is output
+template <class T_dat, class T_ind, data_types T_dat_dt, data_types T_ind_dt>
+class gather8_test_4d : public ::testing::TestWithParam<gather8_test_4d_param> {
```

Comment:
I think we can merge two functions `gather8_test_4d` and `gather8_test_5d` by using `vector` instead of `array`?

Reply:
Done.